generator/src/main/java/com/scottlogic/deg/generator/constraints/StringConstraintsCollection.java

```diff
@@ -0,0 +1,180 @@
+package com.scottlogic.deg.generator.constraints;
```

Comment:
We should consider removing the longer than, shorter than, is length restrictions

and replace them with a single length restriction class that handles all these options and how they interact in one place

Reply:
or, possibly better

this class can do all this handling by internally converting any length based constraints to an allowable lengths, then when merging, simply merge the allowable lengths

Reply:
the set of allowable lengths itself could be an infinite set tho.

Reply:
This elludes to a different implementation that should otherwise yield the same result.